modules/margin-protocol/src/lib.rs

```diff
@@ -443,6 +443,15 @@ impl<T: Trait> Module<T> {
 		<PositionsByTrader<T>>::remove(who, (position.pool, position_id));
 		PositionsByPool::remove(position.pool, (position.pair, position_id));
 
+		// reset trader's equity to $0
+		let count = <PositionsByTrader<T>>::iter(who)
```

Comment:
`let hasPosition = <PositionsByTrader<T>>::iter(who).any(|((p, _), _)| *p == position.pool)`

no need to iterate everything